### PR TITLE
feat: Make isCommunity data fetched from farm auction on chain contract

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

File: src/state/farms/fetchFarmsWithAuctions.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      },
+    ],
+    { requireSuccess: false },
+  )
+  const blocksUntilBlock = auctionData.endBlock.toNumber() - currentBlock
+  if (blocksUntilBlock < 0) {
+    const secondsUntilStart = blocksUntilBlock * BSC_BLOCK_TIME
+    const currentAuctionEndDate = add(new Date(), { seconds: secondsUntilStart })
+    if (
+      Math.abs(now - add(currentAuctionEndDate, { days: FARM_AUCTION_HOSTING_IN_DAYS }).getTime()) /
```
Comment (chefilip) [inline src/state/farms/fetchFarmsWithAuctions.ts:37]: `now` here will no longer be `the now` if `await` functions above are executed slowly.

File: src/state/farms/fetchFarmsWithAuctions.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        (24 * 3600 * 1000) >
+      FARM_AUCTION_HOSTING_IN_DAYS
```
Comment (chefilip) [inline src/state/farms/fetchFarmsWithAuctions.ts:39]: I think using `seconds` format would make it more readable.

File: src/state/farms/fetchFarmsWithAuctions.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        params: [currentAuctionId, 0, 500],
+      },
+    ],
+    { requireSuccess: false },
+  )
+  const blocksUntilBlock = auctionData.endBlock.toNumber() - currentBlock
+  if (blocksUntilBlock < 0) {
+    const secondsUntilStart = blocksUntilBlock * BSC_BLOCK_TIME
+    const currentAuctionEndDate = add(new Date(), { seconds: secondsUntilStart })
+    if (
```
Comment (chefilip) [inline src/state/farms/fetchFarmsWithAuctions.ts:36]: Could we write like this:
```
secondsUntilStart > FARM_AUCTION_HOSTING_IN_SECONDS
```

File: src/state/farms/fetchFarmsWithAuctions.ts (hunk 1 above)
Comment (memoyil) [reply src/state/farms/fetchFarmsWithAuctions.ts]: Done

File: src/state/farms/fetchFarmsWithAuctions.ts (hunk 2 above)
Comment (memoyil) [reply src/state/farms/fetchFarmsWithAuctions.ts]: Done

File: src/state/farms/fetchFarmsWithAuctions.ts (hunk 3 above)
Comment (memoyil) [reply src/state/farms/fetchFarmsWithAuctions.ts]: Done

**Review by chefilip — APPROVED** [review]

**Pull request merged**